### PR TITLE
feat: use aws sdk client

### PR DESCRIPTION
**Pull request by simone-sanfratello**
Refactoring the response process, the first step is to use the standard `sdk` client, because it's expensive to support a custom client, even if adopting `undici` seems to be faster, as you can see https://gist.github.com/simone-sanfratello/41d67d6436269472d766f81ccfab29ab (env node v18).

The following steps will be to be implement prefetching and caching; after that we can evaluate if introduce a custom client - to be used in all the parts/repos of the `e-ipfs`.

Please note I'm adopting a pragmatic approach, I don't have any preferences using `undici` or `sdk` or whatever, I just want to solve issues in the fastest possible way, and that's why I'm refactoring in this way, because I don't see enough advantages to maintain the custom client.

**Note (to me): do not squash merge!**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,11 +9,10 @@ _Variables in bold are required._
 | AWS_ACCESS_KEY_ID     |               | The AWS key ID.                                                          |
 | AWS_REGION            |               | The AWS region.                                                          |
 | AWS_SECRET_ACCESS_KEY |               | The AWS access key.                                                      |
-| CACHE_BLOCK_INFO     | `false`       | Set to `true` to cache block informations with in a in-memory LRU cache. |
-| CACHE_BLOCK_INFO_SIZE | `1000`       | Max entries of the block info cache.                                     |
+| CACHE_BLOCK_INFO      | `true`       | Set to `true` to cache block informations with in a in-memory LRU cache. |
```
Comment (simone-sanfratello) [inline README.md:12]: changing default block cache to `true`

**Review by codyzu — APPROVED** [review]
Makes the code much simpler, so I'm a big fan of that.

Plus your [benchmarks](https://gist.github.com/simone-sanfratello/41d67d6436269472d766f81ccfab29ab) indicate that the s3 client is actually faster, right?

I guess the only downside is if there are issues in the AWS SDK client they might be harder to fix.

Given the importance of S3 and AWS, IMO the simplification of the code is worth the potential risks. Maintainng all of that code that you were able to simplify for months or years seems really valuable.

Do you have a system in place to monitor the performance (and memory consumption) of this change when it's in production? It would be sweet if there was a way to do blue/green or canary testing, but that's usually pretty hard in a service built on native cloud services.

IMO, yes... try this! The code is so much simpler. If there is a catastrophic problem you can always revert.

🥳 Fun Fact: I thought it would be interesting to see the source of the AWS S3 client to understand what they use under the hood if it is more performant. Turns out it's all generated typescript 😱 🤯 . Part of AWS "smithy" project with appears to be written in java 🤣 : https://awslabs.github.io/smithy/2.0/index.html

**Comment by simone-sanfratello** [comment]
> Plus your [benchmarks](https://gist.github.com/simone-sanfratello/41d67d6436269472d766f81ccfab29ab) indicate that the s3 client is actually faster, right?
> 

They don't reveal a big difference between the two, but we have to consider that the sdk client is quite complex, for example just in that code the bare http client need signature implementation

> Do you have a system in place to monitor the performance (and memory consumption) of this change when it's in production? It would be sweet if there was a way to do blue/green or canary testing, but that's usually pretty hard in a service built on native cloud services.

yes, we have metrics to see the real performance in production, if we see perfomance issues, we rollback

> partying_face Fun Fact: I thought it would be interesting to see the source of the AWS S3 client to understand what they use under the hood if it is more performant. Turns out it's all generated typescript scream exploding_head . Part of AWS "smithy" project which appears to be written in java rofl : https://awslabs.github.io/smithy/2.0/index.html

from source

```ts
    /**
     * The HTTP handler to use. Fetch in browser and Https in Nodejs.
     */
    requestHandler?: __HttpHandler;
```

File: test/bitswap-1.0.0.test.js [hunk 2]
```diff
@@ -58,8 +60,6 @@ t.test(`${protocol} - uses the right fields when serializing and deserializing`,
 })
 
 t.test(`${protocol} - type=Block - sendDontHave=true - 2 hits / 2 misses - 2 blocks received`, async t => {
-  t.plan(4)
```
[the remark this answers is not in the document]
Comment (simone-sanfratello) [reply test/bitswap-1.0.0.test.js:61]: using `async t` callback, plan are not needed unless we need to wait for callback or events in the test

File: test/utils/mock.js [hunk 3]
```diff
@@ -1,108 +1,109 @@
...
+  mockDynamoQueryCommand({ table: config.linkTableV1, keyName: config.linkTableBlockKey, keyValue: cidToKey(cid3) })
+  mockDynamoQueryCommand({ table: config.linkTableV1, keyName: config.linkTableBlockKey, keyValue: cidToKey(cid4) })
+  mockDynamoQueryCommand({ table: config.linkTableV1, keyName: config.linkTableBlockKey, keyValue: cidToKey(cid5), response: readBlock('blocks/db-v1/cid5.json') })
+  mockDynamoQueryCommand({ table: config.linkTableV1, keyName: config.linkTableBlockKey, keyValue: cidToKey(cid6), response: readBlock('blocks/db-v1/cid6.json') })
+  mockDynamoQueryCommand({ table: config.linkTableV1, keyName: config.linkTableBlockKey, keyValue: cidToKey(cid7), response: readBlock('blocks/db-v1/cid7.json') })
+  mockDynamoQueryCommand({ table: config.linkTableV1, keyName: config.linkTableBlockKey, keyValue: cidToKey(cid8), response: readBlock('blocks/db-v1/cid8.json') })
+  mockDynamoQueryCommand({ table: config.linkTableV1, keyName: config.linkTableBlockKey, keyValue: cidToKey(cid9), response: readBlock('blocks/db-v1/cid9.json') })
+
+  // // searchCarInDynamoV0
```
[the remark this answers is not in the document]
Comment (simone-sanfratello) [reply test/utils/mock.js:88]: Forgot to remove ^^'

**Pull request merged**